src/main/java/seedu/address/model/AlfredModelManager.java

```diff
@@ -0,0 +1,380 @@
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.logging.Logger;
+
+import javafx.collections.transformation.FilteredList;
+import seedu.address.commons.core.GuiSettings;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.exceptions.AlfredException;
+
+import seedu.address.model.entity.Id;
+import seedu.address.model.entity.Mentor;
+import seedu.address.model.entity.Participant;
+
+import seedu.address.model.entity.Team;
+
+import seedu.address.storage.AlfredStorage;
+
+/**
+ * Represents the in-memory alfredModel of the address book data.
+ */
+public class AlfredModelManager implements AlfredModel {
+    private static final Logger logger = LogsCenter.getLogger(AlfredModelManager.class);
+
+    private final UserPrefs userPrefs;
+    private final Alfred alfred;
+
+    // EntityLists
+    private FilteredList<Participant> participantFilteredList;
+    private FilteredList<Team> teamFilteredList;
+    private FilteredList<Mentor> mentorFilteredList;
+
+    private FilteredList<Participant> filteredParticipantList = null;
+    private FilteredList<Team> filteredTeamList = null;
+    private FilteredList<Mentor> filteredMentorList = null;
+
+    /**
+     * Initializes a AlfredModelManager with the given addressBook and userPrefs.
+     */
+    private AlfredModelManager(ReadOnlyAlfred initialData, ReadOnlyUserPrefs userPrefs) throws AlfredException {
+        super();
+        requireAllNonNull(initialData, userPrefs);
+
+        logger.fine("Initializing with Participant List: " + initialData.getParticipantList());
+        logger.fine("Initializing with Mentor List" + initialData.getMentorList());
+        logger.fine("Initializing with Team List: " + teamFilteredList);
+        logger.fine("Initializing with UserPrefs: " + userPrefs);
+
+        this.userPrefs = new UserPrefs(userPrefs);
+        this.alfred = (Alfred) initialData;
+        this.participantFilteredList = new FilteredList<Participant>(initialData.getParticipantList());
+        this.teamFilteredList = new FilteredList<Team>((initialData.getTeamList()));
+        this.mentorFilteredList = new FilteredList<Mentor>(initialData.getMentorList());
+    }
+
+    /**
+     * Returns a {@code AlfredModelManager} with the data from {@code storage}'s address
+     * book and {@code userPrefs}. <br>
+     * The data from the sample address book will be used instead if
+     * {@code storage}'s address book is not found, or an empty address book will be
+     * used instead if errors occur when reading {@code storage}'s address book.
+     */
+    public static AlfredModel initModelManager(AlfredStorage storage, ReadOnlyUserPrefs userPrefs) throws AlfredException {
+        ReadOnlyAlfred initialData = new Alfred(storage, userPrefs);
+
+        return new AlfredModelManager(initialData, userPrefs);
+    }
+
+    /**
+     * Initializes the various lists used. If storage contains no data, it defaults
```

Comment:
Can I just clarify that this means that the initialisation of the Entity Lists from storage occurs in the initModelManager method?